server/gateway.go

```diff
@@ -45,7 +45,7 @@ const (
 	// hash of origin cluster name and <server> is 6 characters hash of origin server pub key.
 	gwReplyPrefix    = "_GR_."
 	gwReplyPrefixLen = len(gwReplyPrefix)
-	gwHashLen        = 6
+	gwHashLen        = sysHashLen
```

Comment:
Just making a note here that this may break pre GWs between pre 2.2.0 and 2.2.0 servers. Not sure, will have to experiment/dig a bit more.

Reply:
Alternatively, we could have a getHash() with a count and default would be 8 but for GW code, I would call with 6 to maintain compatibility. No other code in gateway.go has been changed it seems, so that would be safer I think.

Reply:
Possibly, need to know if its really a problem.

Reply:
Will check and we can always adjust if it happens to be a problem.